blanderbit/vue3-easy-data-table
# Feature: add possibility to group table rows by columns

# Definition of Done:

- [X] Added the ability to group table rows by columns.
- [X] Added the ability to hide/show child elements of a group.

# Internal changes:

- Updated `table-props.md` file according to new changes.
- Added new hooks:
   - `useGroupBy` - to group table rows.
- Added a new props for table Header:
   - `groupable` (type: `Boolean`, default: `false`) - Indicates if this column can be used to group rows.
   - `grouped` (type: `Boolean`, default: `false`) - Group the rows by this column.
   - `groupBy` (type: `Function`, default: `null`) - This function convert the cell value to another value, means you can change the label for grouped rows. See `table-props.md` for more information.
- Added a new plugin to `eslint` called `unused-imports`, that allows to remove unused imports.
- Fixed eslint issue related to importing types.
- If you group table rows by specific column, that column can not be hidden using table property manager, it should be disabled.
- Type correction for the codebase.

# Testing Done:

1. Run application.
2. Click on group icon in the header cell.
3. You should see the grouped rows by the column:
   - You can sort by grouped column (just click on the sort icon beside to header label).
   - You hide/show child elements of a group (use the corresponding icon to do it, which is to the right of header label
   or just click on header label to toggle visibility of child elements of a group).
5. Ungroup rows using corresponding icon to the right of the group's label.
6. You should see table rows in the usual way.
